src/mlpack/methods/ann/layer/repeat.hpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * @file methods/ann/layer/repeat.hpp
+ * @author Adam Kropp
+ *
+ * Definition of the Repeat class, which repeats the input n times
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_REPEAT_HPP
+#define MLPACK_METHODS_ANN_LAYER_REPEAT_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include "layer.hpp"
+
+namespace mlpack {
+
+/**
+ * Implementation of the Repeat class. The Repeat class repeats the
+ * input a specified number of times along a each dimension.  The output
+ * will have the same number of dimensions as the input, with each dimension
+ * multiplied by a specified multiple.
```

Comment:
```suggestion
 * multiplied by a specified multiple.  The input can be repeated in an
 * interleaved or block fashion, depending on the parameters given to the
 * constructor.
```
Just an idea, might be nice to point this out.